### PR TITLE
Add 0.7.1 Support

### PR DESCRIPTION
**Pull request by SirNexus**
Modified source.ip to be included as an IPv4 rather than string since mixer implemented type constraints.
Updated protocol buffers to Mixer 0.7.1. Also commented some of the code.
Added destination and source label dynamic values support.
Added two new arguments: mixer_source_labels, mixer_destination_labels

### CODE REVIEW

File: mixer-ngx/src/ngx/request.rs [hunk 1]
```diff
@@ -55,10 +54,12 @@ impl MixerConfig for ngx_http_request_s  {
                 // TODO: remove header
                 match name.as_ref()  {
 
-                    SRC_IP_HEADER  => {
-                        ngx_http_debug!(self,"source IP received {}",&value);
-                        attr.insert_string_attribute( SOURCE_IP,&value);
-                    },
+                    // TODO: Update SOURCE_IP to bytes value,
```
Comment (peterkellyonline) [inline mixer-ngx/src/ngx/request.rs]: Prefer not to have uncommented code, better to remove

File: mixer-ngx/src/ngx/request.rs [hunk 2]
```diff
@@ -87,20 +88,22 @@ impl MixerConfig for ngx_http_headers_out_t {
...
         attr.insert_int64_attribute(RESPONSE_SIZE, self.content_length_n);
 
-        // fill in the string value
+        // fill in the string value from all values of request struct
         let mut map: HashMap<String,String> = HashMap::new();
         {
             for (name,value) in self.headers_iterator()   {
-         //       ngx_http_debug!("processing out header name: {}, value: {}",&name,&value);
+
+            //    ngx_http_debug!("processing out header name: {}, value: {}",name,value);
```
Comment (peterkellyonline) [inline mixer-ngx/src/ngx/request.rs]: same comment as above, prefer no uncommented code

File: mixer-ngx/src/ngx/server_config.rs [hunk 3]
```diff
@@ -1,41 +1,90 @@
...
+    pub source_port:            ngx_uint_t,
+    pub source_labels:          service_labels
 }
 
 impl MixerConfig for  ngx_http_mixer_srv_conf_t  {
 
     fn process_istio_attr(&self, attr: &mut AttributeWrapper) {
+        ngx_event_debug!("Processing server attributes");
 
+        ngx_event_debug!("{}: {}",DESTINATION_SERVICE, self.destination_service.to_str());
```
Comment (peterkellyonline) [inline mixer-ngx/src/ngx/server_config.rs]: this debugging makes the code hard to read. I would suggest grouping this together below or above the other code. `to_str()` is called two times for every variable in this approach too.

File: mixer-transport/src/attribute/attr_wrapper.rs [hunk 4]
```diff
@@ -139,6 +148,10 @@ impl AttributeWrapper  {
                 &AttrValue::StringMap(ref str_value) => {
                     attrs.mut_string_maps().insert(index, map_string_map(str_value,  dict));
                 }
+                &AttrValue::Ip(ref ip_value) => {
+                    // ngx_event_debug!("IP value!! String: {}, Bytes vector: {}", ip_value, ip_value.octets().to_vec());
```
Comment (peterkellyonline) [inline mixer-transport/src/attribute/attr_wrapper.rs]: remove commented out code

File: mixer-ngx/src/ngx/server_config.rs [hunk 5]
```diff
@@ -38,53 +38,82 @@ impl MixerConfig for  ngx_http_mixer_srv_conf_t  {
...
+    fn to_map(&self) -> HashMap<String,String> {
 
-    unsafe {
-        if ((*labels).key as *const u8).is_null() {
+        let mut out_map: HashMap<String,String> = HashMap::new();
+
+        if (self.key as *const u8).is_null() {
             return out_map;
+        } else {
+            for (key,value) in self.iter() {
```
Comment (SirNexus) [inline mixer-ngx/src/ngx/server_config.rs]: Added iterator for service_labels struct to more clearly add elements to string map.

**Comment by peterkellyonline** [comment]
LGTM

**Pull request merged**